src/components/ui/members/member-profile-container.tsx

```diff
@@ -0,0 +1,96 @@
+"use client"
+
+import { useState } from "react"
+import { useRouter } from "next/navigation"
+import { useUser, useUserMutation } from "@/lib/api/users"
+import { ProfileInfoUpdateForm } from "@/components/ui/members/profile-info-update-form"
+import { PasswordUpdateForm } from "@/components/ui/members/password-update-form"
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card"
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs"
+import { useAuthStore } from "@/lib/providers/auth-store-provider"
+import { Id } from "@/types/general"
+import { User, NewUserPassword } from "@/types/user"
+import { useUserPasswordMutation } from "@/lib/api/users"
+import { useLogoutUser } from "@/lib/hooks/use-logout-user"
+
+export function MemberProfileContainer({ userId }: { userId: Id }) {
+    const { userId: currentUserId, logout } = useAuthStore((state) => state)
+    const { user, isLoading, refresh } = useUser(userId)
+    const { update: updateUser, isLoading: isUpdating } = useUserMutation()
+    const { update: updatePassword } = useUserPasswordMutation()
+    const [activeTab, setActiveTab] = useState("profile")
+    const logoutUser = useLogoutUser()
+
+    const handleProfileUpdate = async (
+        updatedUser: User
+    ) => {
+        try {
+            await updateUser(userId, updatedUser)
+            refresh()
+        } catch (error) {
+            console.error("Error updating profile:", error)
+        }
+    }
+
+    const handlePasswordUpdate = async (updatedPassword: NewUserPassword) => {
+        try {
+            await updatePassword(userId, updatedPassword)
+            // Logout the user when the password is updated
+            logoutUser()
+        } catch (error) {
+            console.error("Error updating password:", error)
+        }
+    }
+
+    if (isLoading) {
+        return (
+            <div className="flex items-center justify-center h-64">
+                <div className="text-center">
+                    <div className="animate-spin h-8 w-8 border-4 border-primary border-t-transparent rounded-full mx-auto"></div>
+                    <p className="mt-2">Loading user member profile...</p>
```

Comment:
```suggestion
                    <p className="mt-2">Loading member profile...</p>
```

Reply:
"Users" are "Members" in the frontend display layer.

Reply:
Updated as part of 80d1f2f359e20902570496e3bc0734807f793b2b